### PR TITLE
[v2]: drop support for old operating systems

### PR DESCRIPTION
**Pull request by saghul**
Time has come. I've come here to delete code and chew bubblegum. And I'm all out of bubblegum.

🔥 🔥 🔥 Windows XP 🔥 🔥 🔥 

/cc @libuv/collaborators 

I haven't tried to compile this yet, it's possible some includes are missing now that we call many functions directly instead of the dynamically loaded ones.

More improvements might be possible on Windows now that we drop the XP baggage, for example using CancelIoEx, but that's a story for another patch.

I also dropped support for MSVC < 2015, everyone just get with the program already.

### CODE REVIEW

File: src/win/thread.c [hunk 1]
```diff
@@ -371,236 +352,46 @@ int uv_sem_trywait(uv_sem_t* sem) {
...
-  LeaveCriticalSection(&cond->fallback.waiters_count_lock);
-
-  if (have_waiters)
-    SetEvent(cond->fallback.signal_event);
-}
-
-
-static void uv_cond_condvar_signal(uv_cond_t* cond) {
-  pWakeConditionVariable(&cond->cond_var);
+  /* nothing to do */
```
Comment (bnoordhuis) [inline src/win/thread.c]: Can you capitalize and punctuate the comment?

**Comment by bnoordhuis** [comment]
LGTM with a comment.  Also down with dropping support for pre-VS 2015.

One of the commits says "drop support for macOS < 10.9" but it looks like it's actually dropping support for < 10.7?

**Comment by saghul** [comment]
> One of the commits says "drop support for macOS < 10.9" but it looks like it's actually dropping support for < 10.7?

See https://github.com/libuv/libuv/pull/966/commits/2cfc28601053586a4492a186821b3a24996ffcbf#diff-6a16903c26af4b4035eda9922a73ecc9L339

It will probably continue to compile, though with a warning. Same for 10.7, it would compile and run, but a test or 2 will fail. It's kinda more of a statement. I'll also update the README to reflect that.

**Comment by cjihrig** [comment]
Nice! LGTM

**Comment by indutny** [comment]
LGTM if it works :)

**Comment by saghul** [comment]
CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/82/

**Comment by saghul** [comment]
CI is 💚 for the relevant platforms. Game on!

**Comment by saghul** [comment]
Btw, @piscisaureus if you have suggestions on things we could do better now that XP/2k3 support is not a concern, please let me know!

**Pull request merged**